### PR TITLE
Improve accessibility of the textfield and snackbar

### PR DESCRIPTION
**Pull request by pith**
- Add possibility to have a role attribute on `<Text>`
- Add the role "alert" on `<Snackbar>` for accessibility
- Add aria-hidden on the `<Textfield>` clear button

### CODE REVIEW

File: src/text/index.tsx [hunk 1]
```diff
@@ -58,8 +61,10 @@ const Text = ({
     content = replaceNewLineWithBR(children)
   }
 
+  const attrRole = !isEmpty(role) ? { role } : {}
+
   return (
-    <Tag className={cc([baseClassName, displayClassName, className])}>
+    <Tag {...attrRole} className={cc([baseClassName, displayClassName, className])}>
```
Comment (loick) [inline src/text/index.tsx]: you can put role with null as a default value and put it directly on Tag?

File: src/textField/index.tsx [hunk 2]
```diff
@@ -226,6 +230,7 @@ export default class TextField extends PureComponent<TextFieldProps, TextFieldSt
               onClick={buttonOnClick}
               tabIndex="-1"
               title={buttonTitle}
+              aria-hidden={isEmpty(buttonTitle)}
```
Comment (wakooka) [inline src/textField/index.tsx]: is buttonTitle a string? in that case shouldn't you do `{!!buttonTitle}` instead?

Comment (pith) [reply src/textField/index.tsx]: `isEmpty` is ok and is more readable I think.

```
> isEmpty('')
true
> isEmpty('plop')
false
```

File: src/text/index.tsx (hunk 1 above)
Comment (Tuxosaurus) [reply src/text/index.tsx]: If we want to prevent misuses of this component, we might as well have 'alert' as the default value

Comment (pith) [reply src/text/index.tsx]: It's defined as alert for the snackbar, but it shouldn't be the default for the Text component ;)

**Pull request merged**